### PR TITLE
XYLayout and RulerLayout shouldn't throw exception on invalid constraint

### PR DESCRIPTION
**Pull request by ptziegler**
Depends on #439

### CODE REVIEW

**Review by azoitl — APPROVED** [review]

File: org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java [hunk 1]
```diff
@@ -80,8 +86,12 @@ public void layout(IFigure container) {
 	@Override
 	public void setConstraint(IFigure figure, Object newConstraint) {
 		if (newConstraint != null && !(newConstraint instanceof Integer)) {
-			throw new IllegalArgumentException("RulerLayout was given " + newConstraint.getClass().getName() //$NON-NLS-1$
-					+ " as constraint for Figure. Integer expected!"); //$NON-NLS-1$
+			if (!(newConstraint instanceof GridData)) {
```
Comment (azoitl) [inline org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java]: I think this is a copy paste error.

**Comment by ptziegler** [comment]
One second, I think there's a copy-paste error in the RulerLayout...

File: org.eclipse.gef.tests/src/org/eclipse/gef/test/RulerLayoutTests.java [hunk 2]
```diff
@@ -57,9 +77,12 @@ public void testConstraint() {
 	/**
 	 * Use Case: Constraint must be of type int -> error
 	 */
-	@Test(expected = IllegalArgumentException.class)
+	@Test
 	public void testInvalidConstraint() {
```
Comment (azoitl) [inline org.eclipse.gef.tests/src/org/eclipse/gef/test/RulerLayoutTests.java:81]: Do we now need a test where we are giving an Integer and check that no message was logged?

**Review by azoitl — CHANGES_REQUESTED** [review]

**Comment by azoitl** [comment]
> One second, I think there's a copy-paste error in the RulerLayout...

I also just noticed.

File: org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java [hunk 3]
```diff
@@ -80,8 +85,12 @@ public void layout(IFigure container) {
 	@Override
 	public void setConstraint(IFigure figure, Object newConstraint) {
 		if (newConstraint != null && !(newConstraint instanceof Integer)) {
-			throw new IllegalArgumentException("RulerLayout was given " + newConstraint.getClass().getName() //$NON-NLS-1$
-					+ " as constraint for Figure. Integer expected!"); //$NON-NLS-1$
+			if (!(newConstraint instanceof Integer)) {
```
Comment (azoitl) [inline org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java]: I think the if is not needed anymore. we have all the checks in line 87

**Review by azoitl — CHANGES_REQUESTED** [review]

File: org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java (hunk 3 above)
Comment (ptziegler) [reply org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java]: That was the copy-paste error I mentioned 😉

File: org.eclipse.gef.tests/src/org/eclipse/gef/test/RulerLayoutTests.java (hunk 2 above)
Comment (ptziegler) [reply org.eclipse.gef.tests/src/org/eclipse/gef/test/RulerLayoutTests.java:83]: I suppose that's easy enough to do... Done.

File: org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java (hunk 3 above)
Comment (ptziegler) [reply org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java]: In any case, that should also be fixed now.

**Review by azoitl — APPROVED** [review]

**Comment by ptziegler** [comment]
I've also added a null check to the GridLayout class. Otherwise the logger would potentially cause a NPE.

**Comment by azoitl** [comment]
> > I've also added a null check to the GridLayout class. Otherwise the logger would potentially cause a NPE.
> 
> I really shouldn't rush this stuff... Argh.

I know how you feel. Been there done that.

**Comment by ptziegler** [comment]
Alright... I've gone over everything again and I don't see anything off. If I missed something, then so be it.

**Review by azoitl — APPROVED** [review]

**Pull request merged**